2a_model.R

```diff
@@ -101,6 +101,84 @@ p2a_targets_list <- list(
       subset_and_write_zarr(do_and_metab, "2a_model/out/well_observed_trn_val_targets.zarr", p2a_trn_val_sites)
     },
     format="file"
+  ),
+  
+  
+  ## CREATE EQUIVALENT TARGETS FOR "MODERATELY-OBSERVED SITES" ##
+  # char vector of moderately-observed train sites
+  tar_target(
+    p2a_trn_sites_medobs,
+    p2_med_observed_sites[!(p2_med_observed_sites %in% val_sites) & !(p2_med_observed_sites %in% tst_sites)]
+  ),
+  
+  # char vector of moderately-observed val and training sites
+  tar_target(
+    p2a_trn_val_sites_medobs,
+    p2_med_observed_sites[(p2_med_observed_sites %in% p2a_trn_sites_medobs) | (p2_med_observed_sites %in% val_sites)]
+  ),
+  
+  # get moderately-observed sites that we use for trning, but also have data in the val time period
+  tar_target(
+    p2a_trn_sites_w_val_data_medobs,
+    p2_daily_with_seg_ids  %>%
+      filter(site_id %in% p2a_trn_val_sites_medobs,
+             !site_id %in% val_sites,
+             date >= val_start_date,
+             date < val_end_date) %>%
+      group_by(site_id) %>%
+      summarise(val_count = sum(!is.na(do_mean))) %>%
+      filter(val_count > 0) %>%
+      pull(site_id)
+  ),
+  
+  # moderately-observed sites that are trning sites but do not have data in val period
+  tar_target(
+    p2a_trn_only_medobs,
+    p2a_trn_sites_medobs[!p2a_trn_sites_medobs %in% p2a_trn_sites_w_val_data_medobs]
+  ),
+  
+  
+  ## WRITE OUT PARTITION INPUT AND OUTPUT DATA ##
+  # write trn met and seg attribute data to zarr
+  # note - I have to subset before passing to subset_and_write_zarr or else I
+  # get a memory error on the join
+  tar_target(
+    p2a_trn_inputs_medobs_zarr,
+    { 
+      trn_input <- p2a_met_data_w_sites %>%
+        filter(site_id %in% p2a_trn_sites_medobs) %>%
+        inner_join(p2a_seg_attr_w_sites, by = "site_id")
+      subset_and_write_zarr(trn_input, "2a_model/out/med_observed_trn_inputs.zarr")
+    },
+    format="file"
+  ),
+  
+  # write trn and val met and seg attribute data to zarr
+  # note - I have to subset before passing to subset_and_write_zarr or else I
+  # get a memory error on the join
+  tar_target(
+    p2a_trn_val_inputs_medobs_zarr,
+    { 
+      trn_input <- p2a_met_data_w_sites %>%
+        filter(site_id %in% p2a_trn_val_sites_medobs) %>%
+        inner_join(p2a_seg_attr_w_sites, by = "site_id")
+      subset_and_write_zarr(trn_input, "2a_model/out/med_observed_trn_inputs.zarr")
+    },
+    format="file"
+  ),
+  
+  # write trn do data to zarr
+  tar_target(
+    p2a_trn_do_medobs_zarr,
+    subset_and_write_zarr(p2_daily_with_seg_ids, "2a_model/out/med_observed_trn_do.zarr", p2a_trn_sites_medobs),
+    format="file"
+  ),
+  
+  # write trn and val do data to zarr
+  tar_target(
+    p2a_trn_val_do_medobs_zarr,
+    subset_and_write_zarr(p2_daily_with_seg_ids, "2a_model/out/med_observed_trn_do.zarr", p2a_trn_val_sites_medobs),
+    format="file"
```

Comment:
We'll only need _one_ target now: `p2a_med_obs_data` - like this target but with the med_obs_sites:
https://github.com/USGS-R/drb-do-ml/blob/6e0afdcc0aed84b7aae1d3497775ff5835e12faf/2a_model.R#L73